src/Analyzers/Core/Analyzers/AbstractBuiltInCodeStyleDiagnosticAnalyzer.cs

```diff
@@ -81,8 +80,8 @@ internal abstract partial class AbstractBuiltInCodeStyleDiagnosticAnalyzer
     /// <summary>
     /// Constructor for a code style analyzer with a multiple diagnostic descriptors with a code style editorconfig option that can be used to configure each descriptor.
     /// </summary>
-    protected AbstractBuiltInCodeStyleDiagnosticAnalyzer(ImmutableDictionary<DiagnosticDescriptor, IOption2> supportedDiagnosticsWithOptions)
-        : this(supportedDiagnosticsWithOptions.Keys.ToImmutableArray())
+    protected AbstractBuiltInCodeStyleDiagnosticAnalyzer(ImmutableArray<(DiagnosticDescriptor Descriptor, IOption2 Option)> supportedDiagnosticsWithOptions)
```

Comment:
>ImmutableArray<(DiagnosticDescriptor Descriptor, IOption2 Option)> [](http://example.com/codeflow?start=57&length=66)

Here and the method below are the places where the code was previously taking an ImmutableDictionary and converting it's keys into an immutablearray